tsimcne/tsimcne.py

```diff
@@ -585,42 +585,48 @@ def transform(
         else:
             return Y
 
+    @staticmethod
+    def get_image_size_from_dataset(X):
+        if isinstance(X, (str, Path)):
+            raise ValueError(
+                "Dataset X is a path, but self.image_size is None. "
+                "The parameter is required, image size cannot be "
+                "inferred from X this way."
+            )
+        x0 = X[0]
+        if hasattr(x0, "__len__") and len(x0) == 2:
+            sample_img, _lbl = x0
+        else:
+            sample_img = x0
+        if not isinstance(sample_img, PIL.Image.Image):
+            raise ValueError(
+                "The dataset does not return PIL images, "
+                f"got {type(sample_img)} instead."
+            )
+        else:
+            return sample_img.size
+
+    @staticmethod
+    def get_data_transform(self, train_or_test, image_size):
+        # data augmentations for contrastive training
+        if train_or_test:
+            data_transform = get_transforms_unnormalized(
+                size=image_size, setting="contrastive", use_ffcv=self.use_ffcv
+            )
+        else:
+            data_transform = get_transforms_unnormalized(
+                size=image_size, setting="none", use_ffcv=self.use_ffcv
+            )
+        return data_transform
```

Comment:
This is a `@staticmethod` but you pass in `self` and use that.  Should probably have another argument for `use_ffcv`.